### PR TITLE
Fix RackHD issue #195

### PR DESCRIPTION
**Pull request by cgx027**
Fix issue #195 by adding return to promise.

https://github.com/RackHD/RackHD/issues/195

### CODE REVIEW

**Comment by cgx027** [comment]
test this please

**Comment by yyscamper** [comment]
There are still other cases that `startServer` doesn't be returned, please fix them as well. Such as https://github.com/cgx027/on-http/blob/Fix_RachHD_issue_195/spec/lib/services/auth-service-spec.js#L265, https://github.com/cgx027/on-http/blob/Fix_RachHD_issue_195/spec/lib/services/auth-service-spec.js#L109, .etc

:+1: after fix all cases.

**Comment by yyscamper** [comment]
test this please

**Comment by cgx027** [comment]
Thanks @yyscamper, I get those enhanced in the new commit.

File: spec/lib/api/login/login-spec.js [hunk 1]
```diff
@@ -260,9 +264,14 @@ describe('Http.Api.Login', function () {
         });
 
         after('stop server, restore mock and configure',function () {
-            cleanUp();
-            //restore endpoint
-            endpoint.authEnabled = true;
+            return cleanUp().then(function(){
+                return Promise.resolve()
```
Comment (WangWinson) [inline spec/lib/api/login/login-spec.js]: set the `endpoint.authEnabled` directly in `then`, no need to wrap another `Promise`

**Comment by cgx027** [comment]
test this please

File: spec/lib/services/auth-service-spec.js [hunk 2]
```diff
@@ -30,17 +31,22 @@ describe('Auth.Service', function () {
...
         server = new Server(endpoint);
-        server.start();
+        return server.start();
     }
 
     function cleanUp(){
-        server.stop();
-        sandbox.restore();
-        restoreConfig();
+        return Promise.resolve()
```
Comment (WangWinson) [inline spec/lib/services/auth-service-spec.js]: Since `server.stop` already wrapped as Promise, may use `return server.stop().then(...)` directly

**Comment by WangWinson** [comment]
:+1: after some style comment fixed

File: spec/lib/api/login/login-spec.js (hunk 1 above)
Comment (cgx027) [inline spec/lib/api/login/login-spec.js]: Good :+1:

File: spec/lib/services/auth-service-spec.js (hunk 2 above)
Comment (cgx027) [inline spec/lib/services/auth-service-spec.js]: Agreed.

**Comment by cgx027** [comment]
Per @dalebremner in https://github.com/RackHD/RackHD/issues/195, the issue has been fixed with this PR.

**Comment by anhou** [comment]
:+1:

**Pull request merged**